src/qfusion.cpp

```diff
@@ -65,40 +66,61 @@ void QFusion::ApplySingleBit(const complex* mtrx, bool doCalcNorm, bitLenInt qub
     // more expensive than directly applying the gates.
     if (qubitCount < MIN_FUSION_BITS) {
         // Directly apply the gate and return.
+        FlushBit(qubitIndex);
         qReg->ApplySingleBit(mtrx, doCalcNorm, qubitIndex);
         return;
     }
 
+    // If we pass the threshold number of qubits for buffering, we just do 2x2 complex matrix multiplication.
+
+    BitBufferPtr bfr = std::make_shared<BitBuffer>(false, (const bitLenInt*)NULL, 0, mtrx);
+    if ((bitControls[qubitIndex].size() > 0) || !(bfr->CompareControls(bitBuffers[qubitIndex]))) {
+        // Flush the old buffer, if the buffered control bits don't match.
+        FlushBit(qubitIndex);
+    }
+
+    // Now, we're going to chain our buffered gates;
+    BitOp inBuffer(new complex[4]);
+    std::copy(mtrx, mtrx + 4, inBuffer.get());
+    bfr->matrix = Mul2x2(inBuffer, bitBuffers[qubitIndex] == NULL ? NULL : bitBuffers[qubitIndex]->matrix);
+    bitBuffers[qubitIndex] = bfr;
+
+    // Almost all additional methods just wrap operations with buffer flushes, or discard the buffers.
```

Comment:
Should this comment be elsewhere?

Reply:
I'm handling this in the following coverage PR, since it's just one comment.

The methods that are doing all the work are ApplySingleBit and its controlled variants. So, I updated this comment to note that it's every method besides ApplySingleBit and the controlled variants, but this is basically the first and one of the only methods you should have to look at to understand what's going on.